onadata/apps/viewer/parsed_instance_tools.py

```diff
@@ -120,24 +120,29 @@ def get_where_clause(query, form_integer_fields=None,
             if isinstance(query, list):
                 query = query[0]
 
-            if isinstance(query, dict) and '$or' in list(query):
-                or_dict = query.pop('$or')
-
-                for or_query in or_dict:
-                    for k, v in or_query.items():
+            if isinstance(query, dict):
+                if '$or' not in list(query):
+                    for k, v in query.items():
                         if v is None:
                             or_where.extend([u"json->>'{}' IS NULL".format(k)])
```

Comment:
Seems like this functionality would not properly handle queries without the `$or` query in it... Was there a need to change the functionality from how it was earlier?

Reply:
Seems the query is handled in the `_parse_where` function could we extend that function to handle `field_value` when it's `None` instead of looping through the query here perhaps?

Reply:
Sounds good.
Made that change